src/main-oe/sgxlkl_run_oe.c

```diff
@@ -2032,25 +2050,35 @@ int main(int argc, char* argv[], char* envp[])
         pthread_setname_np(sgxlkl_threads[i], "ENCLAVE");
     }
 
-    long exit_status = 0;
-
-    for (int i = 0; i < econf->ethreads; i++)
-    {
-        pthread_join(sgxlkl_threads[i], &return_value);
-        if (i == 0)
-        {
-            exit_status = (long)return_value;
-        }
-    }
-    sgxlkl_host_verbose_raw("\n");
-
-    if (oe_enclave)
+    // Wait for the terminating ethread to exit the enclave
+    pthread_mutex_lock(&terminating_ethread_exited_mtx);
+    ret = pthread_cond_wait(
```

Comment:
For very short-lived enclaves or some pathological scheduler conditions, we can exit before we sleep and then never wake.  We should check `enclave_return_status` before exiting here.

Reply:
Hmm, but even if we check `enclave_return_status` before waiting, there would still be a race condition, no?

Reply:
No, the value is only set with the mutex held, so there is an enforced ordering of either check, mismatch, sleep, set wake, or set, check, don’t sleep.

Reply:
Done.